File: lang_mental_models/python/models/event.py
"""日志领域的数据模型。

整个程序传递的是对象，而不是字符串：
    event.level / event.service / event.message
而不是：
    fields[1] / fields[2] / " ".join(fields[3:])

RawLine 是"还没解析"的原始行（带来源），LogEvent 是"已解析"的结构化事件。
"""

from __future__ import annotations

from dataclasses import dataclass


@dataclass
class RawLine:
    """来自某个来源的一条原始日志文本（未解析）。"""

    source: str
    text: str


@dataclass
class LogEvent:
    """一条已解析的结构化日志事件。

    它是 Parser 的产物，也是 Aggregator / Reporter 的输入。
    """

    source: str
    level: str
    service: str
    message: str

    @property
    def is_error(self) -> bool:
        return self.level == "ERROR"

    @property
    def is_warn(self) -> bool:
        return self.level == "WARN"

    @property
    def is_info(self) -> bool:
        return self.level == "INFO"

File: lang_mental_models/go/internal/model/event.go
// Package model 定义在各 Service 之间流动的「消息」（Message）。
//
// Go 最重要的不是 channel，而是用 struct 明确地描述数据模型：
// RawLine 是「还没解析」的原始行，LogEvent 是「已解析」的结构化事件。
package model

// RawLine 是来自某个来源的一条原始日志文本（未解析）。
type RawLine struct {
	Source string
	Text   string
}

// LogEvent 是一条已解析的结构化日志事件，是流水线中流动的核心消息。
type LogEvent struct {
	Source  string
	Level   string
	Service string
	Message string
}

// IsError 报告该事件是否为 ERROR 级别。
func (e LogEvent) IsError() bool { return e.Level == "ERROR" }

// IsWarn 报告该事件是否为 WARN 级别。
func (e LogEvent) IsWarn() bool { return e.Level == "WARN" }

// IsInfo 报告该事件是否为 INFO 级别。
func (e LogEvent) IsInfo() bool { return e.Level == "INFO" }
